pkg/apis/core/v1alpha1/organizationroletemplate_types.go

```diff
@@ -0,0 +1,225 @@
+}
+
+// OrganizationRoleTemplatePhaseType represents all conditions as a single string for printing by using kubectl commands.
+// +kubebuilder:validation:Ready;NotReady;Unknown;Terminating
+type OrganizationRoleTemplatePhaseType string
+
+// Values of OrganizationRoleTemplatePhaseType.
+const (
+	OrganizationRoleTemplatePhaseReady       OrganizationRoleTemplatePhaseType = "Ready"
+	OrganizationRoleTemplatePhaseNotReady    OrganizationRoleTemplatePhaseType = "NotReady"
+	OrganizationRoleTemplatePhaseUnknown     OrganizationRoleTemplatePhaseType = "Unknown"
+	OrganizationRoleTemplatePhaseTerminating OrganizationRoleTemplatePhaseType = "Terminating"
+)
+
+const (
+	OrganizationRoleTemplateTerminatingReason = "Deleting"
+)
+
+// updatePhase updates the phase property based on the current conditions.
+// this method should be called every time the conditions are updated.
+func (s *OrganizationRoleTemplateStatus) updatePhase() {
+	for _, condition := range s.Conditions {
+		if condition.Type != OrganizationRoleTemplateReady {
+			continue
+		}
+
+		switch condition.Status {
+		case ConditionTrue:
+			s.Phase = OrganizationRoleTemplatePhaseReady
+		case ConditionFalse:
+			if condition.Reason == OrganizationRoleTemplateTerminatingReason {
+				s.Phase = OrganizationRoleTemplatePhaseTerminating
+			} else {
+				s.Phase = OrganizationRoleTemplatePhaseNotReady
+			}
+		case ConditionUnknown:
+			s.Phase = OrganizationRoleTemplatePhaseUnknown
+		}
+		return
+	}
+
+	s.Phase = OrganizationRoleTemplatePhaseUnknown
+}
+
+// OrganizationRoleTemplateConditionType represents a OrganizationRoleTemplateCondition value.
+// +kubebuilder:validation:Ready
+type OrganizationRoleTemplateConditionType string
+
+const (
+	// OrganizationRoleTemplateReady represents a OrganizationRoleTemplate condition is in ready state.
+	OrganizationRoleTemplateReady OrganizationRoleTemplateConditionType = "Ready"
+)
+
+// OrganizationRoleTemplateCondition contains details for the current condition of this OrganizationRoleTemplate.
+type OrganizationRoleTemplateCondition struct {
+	// Type is the type of the OrganizationRoleTemplate condition, currently ('Ready').
+	Type OrganizationRoleTemplateConditionType `json:"type"`
+	// Status is the status of the condition, one of ('True', 'False', 'Unknown').
+	Status ConditionStatus `json:"status"`
+	// LastTransitionTime is the last time the condition transits from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// Reason is the (brief) reason for the condition's last transition.
+	Reason string `json:"reason"`
+	// Message is the human readable message indicating details about last transition.
+	Message string `json:"message"`
+}
+
+// GetCondition returns the Condition of the given condition type, if it exists.
+func (s *OrganizationRoleTemplateStatus) GetCondition(t OrganizationRoleTemplateConditionType) (condition OrganizationRoleTemplateCondition, exists bool) {
```

Comment:
we really do need code generators for this stuff. go temples + formatting is enough; shouldn't be too difficult.